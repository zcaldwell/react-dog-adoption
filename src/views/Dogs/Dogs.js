import React from 'react';
import { useState, useEffect } from 'react';
import { fetchDogs } from '../../services/dogs';
import DogList from '../../components/DogList';

export default function Pets() {
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
    };
    fetchData();
  }, []);

  return (
    <div>
      <DogList dogs={dogs} />
    </div>
  );
}
