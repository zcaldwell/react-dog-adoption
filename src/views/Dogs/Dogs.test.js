import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Dogs from './Dogs';

test('Dogs should display all dogs', async () => {
  const { container } = render(
    <MemoryRouter>
      <Dogs />
    </MemoryRouter>
  );
  await screen.findByText('Meet Barton');
  expect(container).toMatchSnapshot();
});
