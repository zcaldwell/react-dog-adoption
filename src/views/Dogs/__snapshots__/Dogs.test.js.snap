// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dogs should display all dogs 1`] = `
<div>
  <div>
    <div
      class="card-container"
    >
      <a
        href="/dogs/3"
      >
        <div
          class="dogcard"
        >
          <h1>
            Meet 
            Ciri
          </h1>
          <img
            class="card-image"
            src="https://placedog.net/1000/563?id=186"
          />
          <p>
            Ciri
             is a(n) 
            1
             year old 
            Shepard
          </p>
        </div>
      </a>
      <a
        href="/dogs/1"
      >
        <div
          class="dogcard"
        >
          <h1>
            Meet 
            Geralt
          </h1>
          <img
            class="card-image"
            src="https://placedog.net/1000/563?id=77"
          />
          <p>
            Geralt
             is a(n) 
            8
             year old 
            Husky
          </p>
        </div>
      </a>
      <a
        href="/dogs/2"
      >
        <div
          class="dogcard"
        >
          <h1>
            Meet 
            Yennefer
          </h1>
          <img
            class="card-image"
            src="https://placedog.net/1000/563?id=160"
          />
          <p>
            Yennefer
             is a(n) 
            10
             year old 
            Cocker-spaniel
          </p>
        </div>
      </a>
      <a
        href="/dogs/4"
      >
        <div
          class="dogcard"
        >
          <h1>
            Meet 
            Zach
          </h1>
          <img
            class="card-image"
            src="https://placedog.net/1000/563?id=118"
          />
          <p>
            Zach
             is a(n) 
            6
             year old 
            German Shepard
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
